catalog/async/catalog-async-postingestplugin/src/main/java/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPlugin.java

```diff
@@ -0,0 +1,202 @@
+        }
+        return new ProcessRequestImpl(processCreateItems,
+                putPostProcessCompleteFlagAndGet(createResponse.getProperties()));
+    }
+
+    private ProcessRequest<ProcessUpdateItem> createUpdateRequest(UpdateResponse updateResponse) {
+        List<Update> updates = updateResponse.getUpdatedMetacards();
+        List<ProcessUpdateItem> processUpdateItems = new ArrayList<>();
+
+        if (CollectionUtils.isNotEmpty(updates)) {
+            for (Update update : updates) {
+                Metacard oldCard = update.getOldMetacard();
+                Metacard newCard = update.getNewMetacard();
+                ProcessUpdateItem processItem =
+                        new ProcessUpdateItemImpl(getProcessResource(newCard),
+                                newCard,
+                                oldCard,
+                                false);
+                processUpdateItems.add(processItem);
+            }
+        }
+
+        return new ProcessRequestImpl(processUpdateItems,
+                putPostProcessCompleteFlagAndGet(updateResponse.getProperties()));
+    }
+
+    private ProcessRequest<ProcessDeleteItem> createDeleteRequest(DeleteResponse deleteResponse) {
+        List<ProcessDeleteItem> processDeleteItems = new ArrayList<>();
+
+        for (Metacard metacard : deleteResponse.getDeletedMetacards()) {
+            ProcessDeleteItem processDeleteItem = new ProcessDeleteItemImpl(metacard);
+            processDeleteItems.add(processDeleteItem);
+        }
+
+        return new ProcessRequestImpl(processDeleteItems,
+                putPostProcessCompleteFlagAndGet(deleteResponse.getProperties()));
+    }
+
+    private ProcessResource getProcessResource(Metacard metacard) {
+        LOGGER.trace(
+                "Getting process resource from catalog framework for metacard with id:{}, sourceId:{}.",
+                metacard.getId(),
+                metacard.getSourceId());
+
+        ResourceRequest request = new ResourceRequestById(metacard.getId());
+
+        return Security.runAsAdmin(() -> {
+            try {
+                ResourceResponse response = catalogFramework.getResource(request,
+                        metacard.getSourceId());
+                Resource resource = response.getResource();
+
+                ProcessResource processResource = new ProcessResourceImpl(metacard.getId(),
+                        resource.getInputStream(),
+                        resource.getMimeTypeValue(),
+                        resource.getName(),
+                        resource.getSize(),
+                        false);
+
+                return processResource;
+            } catch (IOException | ResourceNotFoundException | ResourceNotSupportedException e) {
+                LOGGER.debug("Unable to get resource id:{}, sourceId:{}.",
+                        metacard.getId(),
+                        metacard.getSourceId(),
+                        e);
+            }
+            return null;
```

Comment:
Sounds good. You can simply add `@Nullable` on the method's return type in the interface. No need to add anything to the comments.